game/src/contrib/item/Item.java

```diff
@@ -0,0 +1,374 @@
+package contrib.item;
+
+import contrib.components.InventoryComponent;
+import contrib.crafting.CraftingIngredient;
+import contrib.crafting.CraftingResult;
+import contrib.crafting.CraftingType;
+import contrib.entities.WorldItemBuilder;
+import contrib.item.concreteItem.*;
+
+import core.Entity;
+import core.Game;
+import core.utils.Point;
+import core.utils.components.MissingComponentException;
+import core.utils.components.draw.Animation;
+import core.utils.logging.CustomLogLevel;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * Abstract class that represents every item in the game.
+ *
+ * <p>Every item has a display name, a description, an inventory animation and a world animation.
+ *
+ * <p>Items are droppable and collectable by default. Dropping an item will create a new {@link
+ * Entity} in the world with the {@link contrib.components.ItemComponent} attached. Collecting an
+ * item will add it to the inventory of the collector. To change this behavior, override the {@link
+ * #drop(Entity, Point)} and {@link #collect(Entity, Entity)}.
+ *
+ * <p>Items have a max stack size and a stack size. The stack size is the amount of items on this
+ * stack. The max stack size is the maximum amount of items that can be on this stack. The default
+ * max stack size is 1 and the default stack size is 1. The values can be changed via the {@link
+ * #stackSize()} and {@link #maxStackSize()} methods.
+ */
+public class Item implements CraftingIngredient, CraftingResult {
+
+    protected static final Animation DEFAULT_ANIMATION =
+            new Animation("animation/missing_texture.png");
+    private static final Logger LOGGER = Logger.getLogger(Item.class.getName());
+    private static final Map<String, Class<? extends Item>> items = new HashMap<>();
+
```

Comment:
schreib da mal nochmal javadoc ran alla "Mapt die Bezeichner in Crafting Rezepten mit realen Klassen"